### ISSUE TITLE
VMWare player v17.01 Host Linux Mint vv20.3 Host Linux Mint v21 - shared folder problem

### ISSUE THREAD

**Issue by dbyy** [vmware/open-vm-tools#637]
### Describe the bug

Hi,
I installed as guest LM v21 under host LM v20.3 with shared folder to the host system (on a WIN10 host the shared folder feature worked).
When trying to access a shared folder (which is on a NAS) I get the message <could not find "/mnt/hgfs/NAS Download">.
I checked, the folder "mnt/hgfs" exists but no mount in it.
I know there was a issue reported for an older version of Debian and I thought I posted it her again with some newer details.
Thanks for any help

### Reproduction steps

<empty>

### Expected behavior

I would expect to have access to the shared folder from the NAS

### Additional context

On the guest /usr/bin/vmware-toolbox-cmd version 11.3.5.31214 (build-18557794 is installed.
![Shared folder error 2023-02-14 11-25-23](https://user-images.githubusercontent.com/8461068/218798983-c79bb2da-d634-41e7-8d8a-7de7b5ad6a8a.png)

**Comment by dbyy** [comment]
Here is my clarification:
HOST:  Linux Mint v20.3
GUEST: Linux Mint v21.0  Result: no shared folders in LM v21 from HOST

In the scenario
HOST: Linux Mint v20.3
GUEST: WIN 10  Result: shared folders from HOST are available

HTH

**Comment by dbyy** [comment]
I just did another installation as follow:

HOST: Linux Mint v20.3
GUEST Linux Mint v20.3

in this scenario the shared folders in the GUEST are available in `/mnt/hgfs/<shared1>`, `/mnt/hgfs/<shared2>` etc.

My conclusion is that there must be a difference when the GUEST is Linux Mint v21.
Any ideas?

TIA

**Comment by dbyy** [comment]
... did some further testing and I discovered is the following.

1. I did my previous test on a cloned VM
2. the shared folder settings were enabled from the 'cloned' vm.
3. at this moment (after cloning) the shared folders were NOT available on the clone (/mnt/hgfs/...)
4. when I changed the vm settings: >options > shared folders > (radio button) disable shared folders  > Save
5. went back into > options > shared folders > (radio button) enable shared folders  > Save
6. navigated to folder `/mnt/hgfs/` and the shared folders were now visible and I could 'bookmark' them in 'nemo'

Seems to me that vmware player 'lost' the shared folders, perhaps during cloning.
Any thoughts?

**Comment by johnwvmw** [comment]
OK, so the setup is the nested virtualization.   

What version of Workstation or PLayer for Windows are you using on the Win 10 host?

You have not indicated the hypervisor software being used in the host Linux Mint v20.3

Is it "kvm", "qemu", "Workstation for Linux", or what ??    and what version??

Are the shared folders in the host LM v20.3 and the guest LM v20.3:
  - the same paths?
  - able to access files on the Windows (primary) host?

**Comment by dbyy** [comment]
I wouldn't called a nested virtualization. 
It's based on a native Linux Mint v20.3 installation
vmware player 17.1 on Linux Mint
shared folders have the path...
HOST: /mnt/NAS1, /mnt/NAS2, /mnt/NAS3 etc.
GUEST: /mnt/hgfs/NAS1, /mnt/hgfs/NAS2, /mnt/hgfs/NAS3 etc.

There is NO windows (primary) host

Can't say which hypervisor is running on the Linux Mint v20.3 - How can I check it?
Thanks

**Comment by johnwvmw** [comment]
Sorry, I mistook the Win10 mention in the original post to imply a Windows host at the top of the configuration.

And you are using VMware's Player 17.1 (for Linux) on the LM v20.3 0 player - the hypervisor.

If you follow the steps 4 thru 6 in your earlier response on the LM v21 guest, are the shared folders (and contents) visible?

If that is the case, the what you are seeing is that the fuse-mount is not run automatically at power-on.  I suspect that the working shared folders on the LM v20.5 guest  would not be working is you power it off and then back on.

Take a look at [Known Issues in open-vm-tools Release Notes](https://github.com/vmware/open-vm-tools/blob/stable-12.1.5/ReleaseNotes.md#knownissues) for a work-around.

Let us know if this works or does not.

Thanks

**Comment by dbyy** [comment]
After some more testing I can confirm every time after re-launch of a GUEST VM (Linux Mint) the shared folders were not available in the guest.
I followed the instruction as you suggested in your previous post (Known Issues in open-vm-tools.....) and modified fstab (on the GUEST). Added `vmhgfs-fuse   /mnt/hgfs    fuse    defaults,allow_other    0    0` .
After that, when re-launching the Linux Mint GUEST, the shared folders were available - hip-hip-hurrah.
Worked like a charm. Problem solved.
Many thanks 'johnwvmw'
